flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputChannel.java

```diff
@@ -300,7 +300,7 @@ protected CheckpointBarrier parseCheckpointBarrierOrNull(Buffer buffer) throws I
 		// reset the buffer because it would be deserialized again in SingleInputGate while getting next buffer.
 		// we can further improve to avoid double deserialization in the future.
 		buffer.setReaderIndex(0);
-		return event.getClass() == CheckpointBarrier.class ? (CheckpointBarrier) event : null;
+		return event instanceof CheckpointBarrier ? (CheckpointBarrier) event : null;
```

Comment:
Do we even have any subclass here?

Reply:
No, but if there will be one then we'll need to update every such place.

Reply:
Reverted to minimize changes in this PR.